### PR TITLE
SUMO-227092: Support terraform for azure event hub log source

### PR DESCRIPTION
**Pull request by yyyttthan99**
PR description: 
Added a new resource sumologic_azure_event_hub_log_source in terraform for the new azure source. 

Tests performed: 
1. Manual test by creating a main.tf and using the dev build provider
- Create
Screenshot 2023-11-08 at 11 43 59 AM

- Update
Screenshot 2023-11-09 at 10 30 56 AM

- Delete
Screenshot 2023-11-09 at 10 35 25 AM
Screenshot 2023-11-09 at 10 35 34 AM

NOTE: There is a TODO item in `azure_event_hub_log_source.html.markdown`. We should add a page for Azure Event Hub Log Source under the section https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/ . Then we need to update this doc once we have that page.

### CODE REVIEW

File: website/docs/r/polling_source.html.markdown [hunk 1]
```diff
@@ -2,7 +2,7 @@
 layout: "sumologic"
 page_title: "SumoLogic: sumologic_polling_source"
 description: |-
-  Provides a Sumologic Polling source. This source is used to import data from various AWS products.
+  Provides a Sumologic Polling source. This source is used to import data from various AWS products as well as Azure Event Hub.
```
Comment (samjsong) [inline website/docs/r/polling_source.html.markdown:5]: I know this is the first Azure product, but from doc perspective maybe this is cleaner? What do you think?
```suggestion
  Provides a Sumologic Polling source. This source is used to import data from various AWS and Azure products.
```

File: website/docs/r/azure_event_hub_log_source.html.markdown [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  name        = "my-collector"
+  description = "Just testing this"
+}
+```
+
+## Argument reference
+
+In addition to the [Common Source Properties](https://registry.terraform.io/providers/SumoLogic/sumologic/latest/docs#common-source-properties), the following arguments are supported:
+
+ - `content_type` - (Required) The content-type of the collected data. Details can be found in the [Sumologic documentation for hosted sources][1].
```
Comment (samjsong) [inline website/docs/r/azure_event_hub_log_source.html.markdown:44]: We should be more prescriptive, `content_type` needs to be `AzureEventHubLog` for this source type

File: website/docs/r/azure_event_hub_log_source.html.markdown [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ - `authentication` - (Required) Authentication details for connecting to the S3 bucket.
```
Comment (samjsong) [inline website/docs/r/azure_event_hub_log_source.html.markdown:45]: connecting to Azure Event Hub, not S3 bucket

File: website/docs/r/azure_event_hub_log_source.html.markdown [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     + `type` - (Required) Must be `AzureEventHubAuthentication`.
+     + `shared_access_policy_name` - (Required) Your shared access policy key name.
+     + `shared_access_policy_key` - (Required) Your shared access policy key.
+ - `path` - (Required) The location to scan for new data.
+     + `type` - (Required) Must be `AzureEventHubPath`.
+     + `namespace` - (Required) The namespace of the event hub. 
+     + `event_hub_name` - (Required) The name of the event hub.
+     + `consumer_group` - (Required) The consumer group of the event hub.
+     + `region` - (Optional) The value can be either "Commercial" for azure or "US Gov" for azure gov. Default is for azure.
```
Comment (samjsong) [inline website/docs/r/azure_event_hub_log_source.html.markdown:54]:
```suggestion
     + `region` - (Optional) The region of the event hub. The value can be either `Commercial` for Azure, or `US Gov` for Azure Government. Defaults to `Commercial`.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,6 @@
 ## 2.27.1 (Unreleased)
+FEATURES:
+* **New Resource:** sumologic_azure_event_hub_log_source (GH-590)
```
Comment (samjsong) [inline CHANGELOG.md:3]: Is there a way to indicate that this feature is in Beta?

Comment (yuting-liu) [reply CHANGELOG.md:3]: I think we usually added description of `XXX is in beta` in markdown file: [website/docs/r/azure_event_hub_log_source.html.markdown](https://github.com/SumoLogic/terraform-provider-sumologic/pull/590/files#diff-0cd1b52b8e26bf7e5dc5bb98f301bba1e784b3698bbc2df8da64f8ae15d95ed1)

File: website/docs/r/azure_event_hub_log_source.html.markdown [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+## Argument reference
+
+In addition to the [Common Source Properties](https://registry.terraform.io/providers/SumoLogic/sumologic/latest/docs#common-source-properties), the following arguments are supported:
+
+ - `content_type` - (Required) The content-type of the collected data. Details can be found in the [Sumologic documentation for hosted sources][1].
+ - `authentication` - (Required) Authentication details for connecting to the S3 bucket.
+     + `type` - (Required) Must be `AzureEventHubAuthentication`.
+     + `shared_access_policy_name` - (Required) Your shared access policy key name.
```
Comment (yuting-liu) [inline website/docs/r/azure_event_hub_log_source.html.markdown:47]: policy name

File: website/docs/r/azure_event_hub_log_source.html.markdown [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+     + `shared_access_policy_key` - (Required) Your shared access policy key.
+ - `path` - (Required) The location to scan for new data.
+     + `type` - (Required) Must be `AzureEventHubPath`.
+     + `namespace` - (Required) The namespace of the event hub. 
+     + `event_hub_name` - (Required) The name of the event hub.
+     + `consumer_group` - (Required) The consumer group of the event hub.
+     + `region` - (Optional) The value can be either "Commercial" for azure or "US Gov" for azure gov. Default is for azure. 
+
+[1]: https://help.sumologic.com/Send_Data/Sources/03Use_JSON_to_Configure_Sources/JSON_Parameters_for_Hosted_Sources
+[2]: https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/XXX
```
Comment (yuting-liu) [inline website/docs/r/azure_event_hub_log_source.html.markdown:57]: Do we have a real azure event hub doc page?

**Comment by yuting-liu** [comment]
1. From your description, it seems that those tests are skipped. From the PR check, I saw azure related tests failed:
```
=== RUN   TestAccSumologicAzureEventHubLogSource_create
    testing.go:705: Step 0 error: errors during apply:
        
        Error: ***
          "status" : 400,
          "id" : "GEZI2-MB312-16K2J",
          "code" : "collectors.validation.fields.missing",
          "message" : "Missing required field: 'resources'"
        ***
        
          on /tmp/tf-test2087671[52](https://github.com/SumoLogic/terraform-provider-sumologic/actions/runs/6804291111/job/18502615457?pr=590#step:5:53)2/main.tf line 7:
          (source code not available)
        
        
--- FAIL: TestAccSumologicAzureEventHubLogSource_create (1.92s)
=== RUN   TestAccSumologicAzureEventHubLogSource_update
    testing.go:705: Step 0 error: errors during apply:
        
        Error: ***
          "status" : 400,
          "id" : "3W1GP-E7KUO-RQUPL",
          "code" : "collectors.validation.fields.missing",
          "message" : "Missing required field: 'resources'"
        ***
        
          on /tmp/tf-test38261[54](https://github.com/SumoLogic/terraform-provider-sumologic/actions/runs/6804291111/job/18502615457?pr=590#step:5:55)101/main.tf line 7:
          (source code not available)
        
        
--- FAIL: TestAccSumologicAzureEventHubLogSource_update (1.94s)
```

2. Besides, did you try running locally built terraform to see if you can perform azure source CRUD operations via terraform? Steps: https://github.com/Sanyaku/codelabs/blob/master/backend/pages/SumoLogicTerraformProvider.md#running-locally-built-terraform-provider

File: website/docs/r/azure_event_hub_log_source.html.markdown (hunk 7 above)
Comment (yyyttthan99) [reply website/docs/r/azure_event_hub_log_source.html.markdown:57]: We don't have it yet, so I listed it as a TODO item in the description. I'm not sure if we should merge this PR first and then update the link or if we shouldn't include the dummy link at all.

File: website/docs/r/polling_source.html.markdown (hunk 1 above)
Comment (yyyttthan99) [reply website/docs/r/polling_source.html.markdown:5]: Sounds reasonable.

File: website/docs/r/polling_source.html.markdown [hunk 8]
```diff
@@ -2,7 +2,7 @@
 layout: "sumologic"
 page_title: "SumoLogic: sumologic_polling_source"
 description: |-
-  Provides a Sumologic Polling source. This source is used to import data from various AWS products.
+  Provides a Sumologic Polling source. This source is used to import data from various AWS and Azure products.
```
Comment (vsinghal13) [inline website/docs/r/polling_source.html.markdown:5]: No need to edit this as this source is deprecated already.

File: website/docs/r/azure_event_hub_log_source.html.markdown [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+     + `shared_access_policy_name` - (Required) Your shared access policy name.
+     + `shared_access_policy_key` - (Required) Your shared access policy key.
+ - `path` - (Required) The location to scan for new data.
+     + `type` - (Required) Must be `AzureEventHubPath`.
+     + `namespace` - (Required) The namespace of the event hub. 
+     + `event_hub_name` - (Required) The name of the event hub.
+     + `consumer_group` - (Required) The consumer group of the event hub.
+     + `region` - (Optional) The region of the event hub. The value can be either `Commercial` for Azure, or `US Gov` for Azure Government. Defaults to `Commercial`.
+
+[1]: https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/XXX
```
Comment (vsinghal13) [inline website/docs/r/azure_event_hub_log_source.html.markdown:56]: We should remove the link as this will be a broken link until a helpdoc page is added.

**Comment by yyyttthan99** [comment]
plox test

**Pull request closed without merging**